lib/practice_exercises.rb

```diff
@@ -1,13 +1,29 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n), linear time, runtime increases as input size increases. 1:1 relation to input size.
+# Space Complexity: O(n), new array created
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  #raise NotImplementedError, "Not implemented yet"
+  new_string = string.downcase.gsub(/[^a-z\s]/i, '')
+  i = 0
+  if new_string[i] == new_string[i - 1]
+    return true
+  else
+    return false
+  end
+  i += 1
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n*m)? 
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
👍 You are returning a substring of the shortest string so the space complexity would be O(m)